src/main/java/org/traccar/protocol/Tlt2hProtocolDecoder.java

```diff
@@ -149,17 +151,20 @@ protected Object decode(
         String status = parser.next();
 
         String[] messages = sentence.substring(sentence.indexOf('\n') + 1).split("\r\n");
+
+        // Filter out end of datagram from messages
+        List<String> filteredMessages = Arrays.stream(messages)
+                .filter(message -> !message.equals("##"))
+                .collect(Collectors.toList());
+
+        messages = filteredMessages.toArray(new String[0]);
+
```

Comment:
This is definitely not what I meant. I think we can filter it out in the `substring` call.

Reply:
I'm not sure how reliable that would be, given the `sentence` doesn't end with `##` in case TCP mode is enabled.
I just pushed a new version, but I'm not sure its better than the initial approach.